yvalenta/lib/client/client.rb

```diff
@@ -0,0 +1,34 @@
+require 'socket'
+
+class Client
+
+  attr_accessor :address, :port
+
+  def initialize(address, port)
+    self.address = address
+    self.port = port
+  end
+
+  def run
+    threads = [Thread.new { read }, Thread.new { write }]
+    threads.each { |t| t.join}
```

Comment:
Space missing inside }.<br>Pass `&:join` as an argument to `each` instead of a block.